From 8804bffbc104bc7fcffe6560acdbac1ba40ea2b5 Mon Sep 17 00:00:00 2001
From: Jan Tulak <jtulak@redhat.com>
Date: Thu, 27 Sep 2018 10:25:19 +0200
Subject: [PATCH 12/22] ssmlib: partitions detecting for loopback devices

Partitions on a loopback device can have different major number
from the loopback device itself. But we know that the partition
has a parent with a certain dev_name, so we can check that dev_name.

Signed-off-by: Jan Tulak <jtulak@redhat.com>
---
 ssmlib/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssmlib/main.py b/ssmlib/main.py
index 07bc038..34a6a70 100644
--- a/ssmlib/main.py
+++ b/ssmlib/main.py
@@ -444,7 +444,7 @@ class DeviceInfo(object):
                 if a == i:
                     continue
                 try:
-                    if dev['major'] != d['major']:
+                    if d['parent_name'] != dev['dev_name'] and dev['major'] != d['major']:
                         continue
                 except KeyError:
                     continue
-- 
2.21.0

